R/tune_args.R

```diff
@@ -0,0 +1,190 @@
+
+
+# @export - lazily and conditionally registered in .onLoad()
+tune_args_recipe <- function(object, full = FALSE, ...) {
+
+  steps <- object$steps
+
+  if (length(steps) == 0L) {
+    return(tune_tbl())
+  }
+
+  res <- purrr::map_dfr(object$steps, tune_args, full = full)
+  tune_tbl(
+    res$name,
+    res$tunable,
+    res$id,
+    res$source,
+    res$component,
+    res$component_id,
+    full = full
+  )
+}
+
+# @export - lazily and conditionally registered in .onLoad()
+tune_args_step <- function(object, full = FALSE, ...) {
+
+  step_id <- object$id
+  # Grab the step class before the subset, as that removes the class
+  step_type <- class(object)[1]
+
+  tune_param_list <- tunable(object)$name
+
+  # remove the non-tunable arguments as they are not important
+  object <- object[tune_param_list]
+
+  # Remove NULL argument steps. These are reserved
+  # for deprecated args or those set at prep() time.
+  object <- object[!purrr::map_lgl(object, is.null)]
+
+  res <- purrr::map_chr(object, find_tune_id)
+  res <- ifelse(res == "", names(res), res)
+
+  tune_tbl(
+    name = names(res),
+    tunable = unname(!is.na(res)),
+    id = unname(res),
+    source = "recipe",
+    component = step_type,
+    component_id = step_id,
+    full = full
+  )
+}
+
+
+# @export - lazily and conditionally registered in .onLoad()
+tune_args_check <- tune_args_step
+
+
+# helpers for tune_args() methods -----------------------------------------
+# these are copied from tune
```

Comment:
These are used in parsnip as well so I plan to look into exporting them in tune and using the exported versions here, instead of the copy.

Reply:
Hmm I am not sure you can export these from tune and use them here.

recipes/parsnip don't rely on tune in any way, and the `tune_args()` methods for model-spec and recipe should be able to run without calling anything from tune.

So these helpers might have to just be duplicated in recipes and parsnip, which is ok

Reply:
The helpers are not used in tune itself any more. So I'd remove them from tune and make the comments in recipes and parsnip point at each other so that we know that the same code is in both packages. Or are the comments overkill?